src/sentry/utils/appleconnect/appstore_connect.py

```diff
@@ -141,42 +138,56 @@ def get_build_info(
-            if type is None or id is None:
+        for relation in page["included"]:
+            rel_type = relation["type"]
+            rel_id = relation["id"]
+            included_relations[(rel_type, rel_id)] = relation
+
+        def get_related(data: JSONData, relation: str) -> Union[None, JSONData]:
+            """Returns related data by looking it up in all the related data included in the page.
+
+            This first looks up the related object in the data provided under the
+            `relationships` key.  Then it uses the `type` and `id` of this key to look up
+            the actual data in `included_relations` which is an index of all related data
+            returned with the page.
+
+            If the `relation` does not exist in `data` then `None` is returned.
+            """
+            rel_ptr_data = safe.get_path(data, "relationships", relation, "data")
+            if rel_ptr_data is None:
+                # The query asks for both the appStoreVersion and preReleaseVersion
+                # relations to be included.  However for each build there is only one of
+                # these that will have the data with type and id, the other will have None
+                # for data.
                 return None
-            return included_relations.get((type, id))
-
-        for build in safe.get_path(page, "data", default=[]):
-            related_appstore_version = get_related(
-                safe.get_path(build, "relationships", "appStoreVersion")
-            )
-            related_prerelease_version = get_related(
-                safe.get_path(build, "relationships", "preReleaseVersion")
-            )
-            related_version = related_appstore_version or related_prerelease_version
-            if not related_version:
-                logger.error("Missing related version for AppStoreConnect `build`")
-                continue
-            platform = safe.get_path(related_version, "attributes", "platform")
-            version = safe.get_path(related_version, "attributes", "versionString")
-            build_number = safe.get_path(build, "attributes", "version")
-            if platform is not None and version is not None and build_number is not None:
+            rel_type = rel_ptr_data["type"]
+            rel_id = rel_ptr_data["id"]
+            return included_relations[(rel_type, rel_id)]
+
+        for build in page["data"]:
+            try:
+                related_appstore_version = get_related(build, "appStoreVersion")
+                related_prerelease_version = get_related(build, "preReleaseVersion")
+                related_version = related_appstore_version or related_prerelease_version
+                if not related_version:
+                    logger.error("Missing related version for AppStoreConnect `build`")
+                    continue
+                platform = related_version["attributes"]["platform"]
+                version = related_version["attributes"]["version"]
```

Comment:
So this is the only real change here?
I did test this locally and had it working. Maybe I did some more refactoring before committing.
Is the field named like that for both `appStoreVersion` and `preReleaseVersion`?

Reply:
😱  that was an excellent question!

https://developer.apple.com/documentation/appstoreconnectapi/appstoreversion/attributes
https://developer.apple.com/documentation/appstoreconnectapi/prereleaseversion/attributes

And indeed the reason this error happened.  let's try this again.